### DISCUSSION TITLE
How to utilise Github cache action for automated builds

### DISCUSSION THREAD

**Discussion by jjenzz** [parcel-bundler/parcel#5068] [General] [answered]
We have a monorepo using Parcel 2 to build our packages and the automated build (via Github Actions) is getting painfully slow. We were able to utilise [the cache action](https://github.com/actions/cache) to speed up steps like dependency installation and I could write a verbose action that [only builds packages that have changed](https://github.com/MarceloPrado/has-changed-path) but I was curious if I could pair the cache action with the `parcel-cache` instead somehow? 

It needs to know when to invalidate the cache if things change so in the case of dependency installation it will [hash the `yarn.lock` file as the identifier](https://github.com/actions/cache/blob/main/examples.md#node---yarn). I'm not sure what changes will require the `.parcel-cache` to be invalidated and there also doesn't appear to be a consistent file we could hash.

Is this currently possible?

**Comment by mischnic** [comment, marked as answer]
The goal is that Parcel automatically invalidates anything that needs to be invalidated. But we're not quite there yet

**Reply by jjenzz** [reply]
Hi @mischnic,

Thanks for your speedy reply 🙂 

So, in theory I would be able to tell the cache action where to find the `parcel-cache` and leave it at that? No need to tell it when things change?

**Reply by mischnic** [reply]
Yes, that is the plan. 

The main (known) missing cases are:
- includedFiles (e.g. SCSS imports): https://github.com/parcel-bundler/parcel/issues/3708
- resolver (e.g. updating node_modules): https://github.com/parcel-bundler/parcel/issues/5032, should be fixed rather soon
- env vars (.env files): https://github.com/parcel-bundler/parcel/issues/1625 / https://github.com/parcel-bundler/parcel/issues/4264
